Array_Functions/Upper_Bound/upper_bound.cpp

```diff
@@ -0,0 +1,29 @@
+#include <iostream>
+#include <vector>
+using namespace std;
+
+int upperBound(vector<int>& a, int key) {
+    int l=0, r=a.size()-1;
+    int ans = r+1;
+    while(l<=r) {
+        int mid = (l+r)/2;
```

Comment:
Good, your solution is correct. Only a slight modification is required, to handle cases when 'n' is significantly large. In that case if you calculate mid = (low + high)/2, then the value may overflow. Instead, use mid = (high - low)/2 + low. This is the most standard practice which is used.

Modify this in all three files

Reply:
All right, updated all files